### PR TITLE
DE-2902: Artifact namespace validation

### PR DESCRIPTION
**Pull request by cristiszabo**

### CODE REVIEW

File: ts_sdk/cli/__put_cmd.py [hunk 1]
```diff
@@ -52,6 +56,15 @@ def __folder_type(arg_value):
     raise argparse.ArgumentTypeError('Not valid folder path provided!')
 
 
+def __namespace_type(arg_value: str,
+                     pat=re.compile("^private-[0-9a-zA-Z-]+$")) -> str:
+    if not re.match(pat, arg_value):
+        raise argparse.ArgumentTypeError(
+            f"Invalid namespace {arg_value}. Namespace should look like "
+            "'private-<alphanumeric-and-hyphen-characters>'.")
```
Comment (wkopen) [inline ts_sdk/cli/__put_cmd.py:64]: You can say it should have `private-` as the prefix and then follow by alphanumeric or hyphens. For example, `private-a-b-c-123`. Btw, this makes me think we should not allow the last character be a hyphen. Can you add that check Cristian?

File: ts_sdk/cli/__put_cmd.py [hunk 2]
```diff
@@ -52,6 +56,15 @@ def __folder_type(arg_value):
     raise argparse.ArgumentTypeError('Not valid folder path provided!')
 
 
+def __namespace_type(arg_value: str,
```
Comment (wkopen) [inline ts_sdk/cli/__put_cmd.py:59]: Can we add some unit tests to this function?

**Review by wkopen — CHANGES_REQUESTED** [review]
Looks good overall. Need unit tests and a clearer error message

File: ts_sdk/cli/__put_cmd.py (hunk 1 above)
Comment (cristiszabo) [reply ts_sdk/cli/__put_cmd.py:64]: Sure. Also, are namespaces like "private---a-b" valid?

File: ts_sdk/cli/__put_cmd.py (hunk 2 above)
Comment (cristiszabo) [reply ts_sdk/cli/__put_cmd.py:59]: Yes, should I only add tests for __namespace_type or all validation functions in __put_cmd.py file?

Comment (wkopen) [reply ts_sdk/cli/__put_cmd.py:59]: Just this function is good.

File: ts_sdk/cli/__put_cmd.py (hunk 1 above)
Comment (wkopen) [reply ts_sdk/cli/__put_cmd.py:64]: good point, we should only allow 1 hyphen

File: __tests__/unit/test_cli_put.py [hunk 3]
```diff
@@ -2,18 +2,46 @@
...
+
+
+@pytest.mark.parametrize("namespace", [
+    # invalid namespaces
+    "nonprivate-test-namespace",
+    "common-org-123",
+    "private-t3st-n@m3sp@ce",
+    "private-a-b-c-",
+    "private---a-b",
+    "private-my--namespace"
```
Comment (wkopen) [inline __tests__/unit/test_cli_put.py:43]: others can be `-` at the beginning and the end of the string

File: __tests__/unit/test_cli_put.py [hunk 4]
```diff
@@ -2,18 +2,46 @@
...
     assert args.auth_token == 'cfg-token'
     assert args.ignore_ssl == True
+
+
+@pytest.mark.parametrize("namespace", [
+    # valid namespaces
+    "private-123",
+    "private-a-b-c-1",
+    "private-my-namespace-123",
+    "private-Test-Namespace"
```
Comment (wkopen) [inline __tests__/unit/test_cli_put.py:29]: numbers in the middle?

**Review by wkopen — APPROVED** [review]
Some more test cases I think you can add. Then you are good to merge.

File: __tests__/unit/test_cli_put.py (hunk 4 above)
Comment (cristiszabo) [reply __tests__/unit/test_cli_put.py]: updated

File: __tests__/unit/test_cli_put.py (hunk 3 above)
Comment (cristiszabo) [reply __tests__/unit/test_cli_put.py]: updated

**Pull request merged**